pkg/runctl/controller_test.go

```diff
@@ -187,86 +187,152 @@ func getAPIPipelineRun(cf *fake.ClientFactory, name, namespace string) (*api.Pip
 }
 
 func Test_Controller_syncHandler_delete(t *testing.T) {
-	for _, test := range []struct {
-		name                  string
-		runManagerExpectation func(*runmocks.MockManager)
-		hasFinalizer          bool
-		expectedError         bool
-		expectedFinalizer     bool
-		expectedResult        api.Result
-		expectedState         api.State
-	}{
-
-		{name: "delete with finalizer ok",
-			runManagerExpectation: func(rm *runmocks.MockManager) {
-				rm.EXPECT().Cleanup(gomock.Any()).Return(nil)
+	for _, currentState := range []api.State{
+		api.StateNew,
+		api.StateWaiting,
+		api.StatePreparing,
+		api.StateRunning,
+		api.StateCleaning,
+		api.StateUndefined,
+	} {
+
+		expectedStateOnError := currentState
+		if currentState == api.StateUndefined {
+			expectedStateOnError = api.StateNew
+		}
+
+		for _, test := range []struct {
+			name                  string
+			runManagerExpectation func(*runmocks.MockManager)
+			hasFinalizer          bool
+			expectedError         bool
+			expectedFinalizer     bool
+			expectedResult        api.Result
+			expectedState         api.State
+		}{
+
+			{name: "delete with finalizer ok",
+				runManagerExpectation: func(rm *runmocks.MockManager) {
+					rm.EXPECT().Cleanup(gomock.Any()).Return(nil)
+				},
+				hasFinalizer:      true,
+				expectedError:     false,
+				expectedFinalizer: false,
+				expectedResult:    api.ResultDeleted,
```

Comment:
For my understanding, is this result type known to downstream components (i.e. build-syncer)? How is this result handled?